mix/mix_protocol.nim

```diff
@@ -287,7 +288,7 @@ proc anonymizeLocalProtocolSend*(
 
   var i = 0
   while i < L:
-    if destIsExit(proto) and i == L - 1:
+    if exitNodeIsDestination and i == L - 1:
```

Comment:
What's the thread of logic in terms of setting up/interpreting whether or not exit is/isn't dest?

I think what I'm trying to ask is something along the lines of "what are the sources of  points/developer friction" with respect to handling the complexity introduced by the notion that an it may be one way or the other. 

What's what I see implied in this change is that a forwarding exit _was_ tightly coupled to the selected protocol, and that's changing to something that is specified within the libp2p implementation (e.g. nim-libp2p).  Is it the case that the intent is to provide the upstream impl the means to choose _how_ its own protocol impl is to be mixified?

Reply:
Yes, the top level application, when creating the connection should decide how will the protocol be mixified, as there is no way for mix to predict whether the exit node should be the destination or not, without unnecessarily making mix be aware of upper layer protocol behavior. 
```nim
 let conn = mixProto.toConnection(
    receiverAddr
    receiverPeerId,
    theProtocolId,
    true,  <-------------------------- exitNodeIsDest (default false)
  )
```